sources/react/index-header-individual-component.jsx

```diff
@@ -0,0 +1,129 @@
+/** @ssr-ready **/
+
+/* !!!
+IF YOU ARE EDITING gridicon/index.jsx
+THEN YOU ARE EDITING A FILE THAT GETS OUTPUT FROM THE GRIDICONS REPO!
+DO NOT EDIT THAT FILE! EDIT index-header.jsx and index-footer.jsx instead
+OR if you're looking to change now SVGs get output, you'll need to edit strings in the Gruntfile :)
+!!! */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+// We don't want the icon prop in to otherProps,
+// as we'll pass those to the svg component.
+// That's why we destructure it from the props but not use it.
+export default function({
+	size = 24,
+	onClick,
+	icon: iconProp,
+	className,
+	...otherProps
+}){
+
+	const calculateNeedsOffset = ( icon, size ) => {
+		const iconNeedsOffset = [
+			'gridicons-add-outline',
+			'gridicons-add',
+			'gridicons-align-image-center',
+			'gridicons-align-image-left',
+			'gridicons-align-image-none',
+			'gridicons-align-image-right',
+			'gridicons-attachment',
+			'gridicons-bold',
+			'gridicons-bookmark-outline',
+			'gridicons-bookmark',
+			'gridicons-calendar',
+			'gridicons-cart',
+			'gridicons-create',
+			'gridicons-custom-post-type',
+			'gridicons-external',
+			'gridicons-folder',
+			'gridicons-heading',
+			'gridicons-help-outline',
+			'gridicons-help',
+			'gridicons-history',
+			'gridicons-info-outline',
+			'gridicons-info',
+			'gridicons-italic',
+			'gridicons-layout-blocks',
+			'gridicons-link-break',
+			'gridicons-link',
+			'gridicons-list-checkmark',
+			'gridicons-list-ordered',
+			'gridicons-list-unordered',
+			'gridicons-menus',
+			'gridicons-minus',
+			'gridicons-my-sites',
+			'gridicons-notice-outline',
+			'gridicons-notice',
+			'gridicons-plus-small',
+			'gridicons-plus',
+			'gridicons-popout',
+			'gridicons-posts',
+			'gridicons-scheduled',
+			'gridicons-share-ios',
+			'gridicons-star-outline',
+			'gridicons-star',
+			'gridicons-stats',
+			'gridicons-status',
+			'gridicons-thumbs-up',
+			'gridicons-textcolor',
+			'gridicons-time',
+			'gridicons-trophy',
+			'gridicons-user-circle',
+			'gridicons-reader-follow',
+			'gridicons-reader-following'
+		];
```

Comment:
I think we can shrink individual `*.jsx` file sizes even further if we don't include all the string arrays found here but rather move the logic found here to `grunt-tasks/svg-to-react.js`, which will allow us to only include the readily computed class names in the output files.

Reply:
What's even worse is that these lists are duplicated from `./index.jsx`. Redundancy === maintenance nightmare for everybody who's going to add a new icon. So we should really centralize this information and build individual files statically.

Reply:
Good suggestion!

I'm glad that you mention the "adding new icon" task as the driving use case for this. At the time it wasn't evident to me how to centralize all that logic without introducing a lot of complexity for that particular use case - and that's the reason I stopped to hear my colleagues' opinions first. I think I may have an idea to solve that.